browser/ui/webui/webcompat_reporter/webcompat_reporter_ui.cc

```
@@ -185,14 +207,12 @@ void WebcompatReporterDOMHandler::RegisterMessages() {
 
 void WebcompatReporterDOMHandler::HandleCaptureScreenshot(
     const base::Value::List& args) {
-  auto* render_widget_host_view =
-      web_ui()->GetWebContents()->GetTopLevelRenderWidgetHostView();
-  CHECK(render_widget_host_view);
+  CHECK(render_widget_host_view_);
```

we have a lot of null checks in the helper function. wdyt about just failing silently (omitting the screenshot from the report) instead of enforcing a `CHECK` here?

---

Failing silently and not showing the screenshot to the user sounds fine, but we want to make sure to capture the error on the backend, so let's send an error message in the webcompat report.

---

changed it, also I changed front-en code to not show `View screenshot` link if RenderWidgetHostView is nullptr

---

Having a dedicated error field is fine but I want to also have the screenshot field indicate that there was an error

---

perhaps we should call `DumpWithoutCrashing` instead. it might be easier to monitor these sort of errors if they are in backtrace

---

>I want to also have the screenshot field indicate that there was an error
@ShivanKaul We allow user to mark checkbox and just don't show `View screenshot` link for preview.
And for back-end: webcompat report JSON will be extended by dedicated field (I mentioned above: `"webcompatReportErrors": [ "Could not create screenshot, RenderWidgetHostView unavailable" ]`) to let us know that screenshot has not been created because of RenderWidgetHostView is `nullptr`. 
And also we can add `DumpWithoutCrashing`, like @DJAndries said.
Does it make sense?

---

yep that sounds fine